script/sp1-e2e.sh

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env bash
+
+# Any error will result in failure
+set -e
+
+network=$1  
+block=$2  
+
+# Check if both network and block are set  
+if [ -z "$network" ] || [ -z "$block" ]; then  
+    echo "Error: Both 'network' and 'block' arguments must be provided."  
+    echo "Usage: $0 <network> <block>"  
+    exit 1  
+fi  
+
+# Construct the input file name  
+input_filename="input-${network}-${block}.json"  
+
+# Define the path where the input file is located  
+input_path="./provers/sp1/contracts/src/fixtures"  
+
+# Create the proofParam JSON string using a here-document  
+proofParam=$(cat <<EOF  
+    "proof_type": "native",  
+    "native": {  
+        "json_guest_input": "${input_path}/${input_filename}"  
+    }  
+EOF
+)
+
+# Output the proofParam JSON  
+echo "$proofParam"
+
+# Function to be called on script exit
+cleanup() {
+    echo "Stopping the background server"
+    kill $SERVER_PID
+}
+
+# Trap exit signals to call cleanup
+trap cleanup EXIT
+
+# Make sure required artifacts are built
+# $ make install 
+# $ make build
+echo "Running Sp1 prover"
+nohup cargo run > server.log 2>&1 &
+sleep 10
+
+# Capture the server process ID
+SERVER_PID=$!
+
+# Get the directory of the current script 
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+
+# Function to check the status of prove-block.sh  
+check_prove_block_status() {  
+    RESPONSE=$("$SCRIPT_DIR/prove-block.sh" $network native $block $block "$proofParam")  
```

Comment:
Shouldn't this call the sp1 prover? And then the only data that should be used in the verifier is the data return in the proof data that is returned. Now it goes through an input.json file and then an SP1 prover is run in `gen_verifier` (but that one does not use the raiko sp1 prover implementation either). Not sure why these different intermediate steps are used to test. Something like seems more like how things will be used in practice:
- prove-block.sh sp1 ...
- wait on proof data from prove-block
- verify proof data in contract (which in the current tests means generating a fixture from that proof data).